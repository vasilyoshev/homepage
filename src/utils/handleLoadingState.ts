import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
