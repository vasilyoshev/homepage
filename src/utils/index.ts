export * from './handleLoadingState';
export * from './testUtils';
