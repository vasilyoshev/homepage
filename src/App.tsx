import React, { useEffect } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Home, About } from 'Routes';
import { Header } from 'components';
import { handleLoadingState } from 'utils';
import './App.scss';

export const App: React.FC = () => {
  useEffect(() => {
    handleLoadingState();
  }, []);

  return (
    <Router>
      <Header />

      <Switch>
        <Route path="/about" component={About} />
        <Route exact path="/" component={Home} />
      </Switch>
    </Router>
  );
};

