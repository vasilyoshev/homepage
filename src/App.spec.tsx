import { render } from 'utils';
import { App } from './App';

describe('App', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});
