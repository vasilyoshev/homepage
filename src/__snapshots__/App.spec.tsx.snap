// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders successfully 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <h6
        class="MuiTypography-root title MuiTypography-h6"
      >
        <a
          aria-current="page"
          class="selected"
          href="/"
        >
          Homepage Widgets
        </a>
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <a
            href="/about"
          >
            About
          </a>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
  <div
    class="gridContainer"
  >
    <button
      class="MuiButtonBase-root MuiFab-root addWidgetFab MuiFab-primary"
      tabindex="0"
      title="Add Widget"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
